### PR TITLE
Adds IVFReader

### LINKED ISSUE #636: IVF Reader
## Summary

We have IVF Writer, but IVF Reader also makes sense.

## Motivation

Sometimes we need to read/broadcast the RAW VP8 video to  to peer.

## Describe alternatives you've considered

Add new class/package IVFReader.

## Additional context

None.

### PR DESCRIPTION
**Pull request by justinokamoto**
#### Description

We have an IVF writer but no reader. This can be helpful for some of the use cases described in #636.

#### Reference issue

Fixes #636

### CODE REVIEW

**Comment by justinokamoto** [comment]
Implementation is complete. Planning to write some more tests before having it reviewed.

File: pkg/media/ivfreader/ivfreader.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	}
+
+	payload := make([]byte, header.frameSize)
+	bytesRead, err := i.stream.Read(payload)
+	if err != nil {
+		return nil, nil, err
+	} else if bytesRead != int(header.frameSize) {
+		return nil, nil, fmt.Errorf("incomplete frame data")
+	}
+	return &payload, header, nil
```
Comment (Sean-Der) [inline pkg/media/ivfreader/ivfreader.go]: You can just return `[]byte` I believe. It will not do a copy https://blog.golang.org/go-slices-usage-and-internals#TOC_4.

File: pkg/media/ivfreader/ivfreader.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+func NewWith(in io.Reader) (*IVFReader, *IVFFileHeader, error) {
+	if in == nil {
+		return nil, nil, fmt.Errorf("stream is nil")
+	}
+
+	reader := &IVFReader{
+		stream: in,
+	}
+
+	// TODO: Return file header somewhere else?
```
Comment (Sean-Der) [inline pkg/media/ivfreader/ivfreader.go]: Maybe make the header a member of `IVFReader` maybe? Can make it non-public, and be a method.

Personally I think it is fine to return it. Users can just do `_` and ignore it if they want!

File: pkg/media/ivfreader/ivfreader_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		payload, _, err := reader.ParseNextFrame()
+		assert.Nil(err, "Shouldn't receive parsing errors")
+		if payload == nil {
+			break
+		}
+		framesParsed++
+	}
+	assert.Equal(250, framesParsed, "Should have parsed 250 frames")
+}
+
```
Comment (Sean-Der) [inline pkg/media/ivfreader/ivfreader_test.go]: Can you add a test to assert that the bodies of the frames are right? Just to make sure no off-by-one errors or things like that in the parsing!

**Comment by Sean-Der** [comment]
This is awesome :) 

@justinokamoto I would like to put you in charge of all IVF Reading development. Senior developer position of course.

File: pkg/media/ivfreader/ivfreader.go (hunk 2 above)
Comment (justinokamoto) [reply pkg/media/ivfreader/ivfreader.go]: Sounds good! I’ll keep it as-is then.

**Comment by justinokamoto** [comment]
@Sean-Der Wow, what an honor! Although the title Senior "IVF" Engineer may be slightly misleading 🤔...🧪🧬👶

**Comment by lightsofapollo** [comment]
💯 This is exactly what I was looking for :)

**Comment by Sean-Der** [comment]
Squashed and merged with 03308449a78e4671f029b6503d77df1f0d7d579d

thank you so much @justinokamoto, this is going to make a lot of people happy :)
